src/Yoti.Auth/YotiClientEngine.cs

```diff
@@ -43,17 +43,31 @@ public async Task<ActivityDetails> GetActivityDetailsAsync(string encryptedConne
 
             Dictionary<string, string> headers = HeadersFactory.Create(keyPair, httpMethod, endpoint, httpContent: null);
 
+            Uri requestUri = CreateRequestUri(apiUrl, endpoint);
+
             Response response = await _httpRequester.DoRequest(
                 _httpClient,
                 HttpMethod.Get,
-                new Uri(
-                    apiUrl + endpoint),
+                requestUri,
                 headers,
                 httpContent).ConfigureAwait(false);
 
             return ProfileParser.HandleResponse(keyPair, response);
         }
 
+        private static Uri CreateRequestUri(Uri apiUrl, string endpoint)
+        {
+            string baseApiUrl = apiUrl.ToString();
+            if (baseApiUrl.EndsWith("/", StringComparison.OrdinalIgnoreCase))
+            {
+                return new Uri(baseApiUrl.TrimEnd('/') + endpoint);
```

Comment:
Do we need to check for a trailing slash before calling `TrimEnd('/')`? It seems like we can always trim `baseApiUrl ` based on the docs:
>If no characters can be trimmed from the current instance, the method returns the current instance unchanged.

So the resulting code would be:
```c#
private static Uri CreateRequestUri(Uri apiUrl, string endpoint)
{
    return new Uri(apiUrl.ToString().TrimEnd('/')  + endpoint);
}
```

Reply:
Good spot, have changed in 5b31a24